pkgs/development/tools/godot/default.nix

```diff
@@ -1,103 +1,146 @@
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  sconsFlags =
+    let
+      toSconsString = x: if builtins.isBool x then lib.boolToString x else toString x;
+    in
+      lib.mapAttrsToList (k: v: "${k}=${toSconsString v}") self.passthru.sconsFlagSet;
 
-  outputs = [ "out" "dev" "man" ];
+  shouldInstallBin = true;
+  shouldInstallShortcut = true;
+  shouldInstallHeaders = true;
+  shouldInstallManual = true;
+
+  outputs = []
+    ++ lib.optional (self.shouldInstallBin || self.shouldInstallShortcut) "out"
+    ++ lib.optional self.shouldInstallHeaders "dev"
+    ++ lib.optional self.shouldInstallManual "man";
+
+  # Where to install the godot bin files within the out directory.
+  binInstallPath = "bin";
+
+  # Pattern identifying the godot executable resulting from the scons build.
+  builtGodotBinNamePattern = "godot.*";
+
+  # Name to give the godot executable when installing in the out path.
+  installedGodotBinName = self.pname;
 
   installPhase = ''
-    mkdir -p "$out/bin"
-    cp bin/godot.* $out/bin/godot
+    runHook preInstall
+
+    if [ $shouldInstallBin ]; then
+      echo "Installing godot binaries."
+      bin="$out/$binInstallPath"
+      mkdir -p "$bin"
+      cp -R bin/. "$bin"
+      mv "$bin"/$builtGodotBinNamePattern "$bin/$installedGodotBinName"
+    fi
+
+    if [ $shouldInstallShortcut ]; then
+      echo "Installing godot shortcut."
+      share="$out"/share
+      mkdir -p "$share/"{applications,icons/hicolor/scalable/apps}
+      cp misc/dist/linux/org.godotengine.Godot.desktop "$share"/applications
+      cp icon.svg "$share"/icons/hicolor/scalable/apps/godot.svg
+      cp icon.png "$share"/icons/godot.png
+      substituteInPlace "$share"/applications/org.godotengine.Godot.desktop --replace "Exec=godot" "Exec=\"$out/bin/$pname\""
+    fi
 
-    mkdir "$dev"
-    cp -r modules/gdnative/include $dev
+    if [ $shouldInstallHeaders ]; then
+      echo "Installing godot headers."
+      mkdir -p "$dev"
+      cp -R modules/gdnative/include "$dev"
+    fi
 
-    mkdir -p "$man/share/man/man6"
-    cp misc/dist/linux/godot.6 "$man/share/man/man6/"
+    if [ $shouldInstallManual ]; then
+      echo "Installing godot manual."
+      installManPage misc/dist/linux/godot.6
+    fi
 
-    mkdir -p "$out"/share/{applications,icons/hicolor/scalable/apps}
-    cp misc/dist/linux/org.godotengine.Godot.desktop "$out/share/applications/"
-    cp icon.svg "$out/share/icons/hicolor/scalable/apps/godot.svg"
-    cp icon.png "$out/share/icons/godot.png"
-    substituteInPlace "$out/share/applications/org.godotengine.Godot.desktop" \
-      --replace "Exec=godot" "Exec=$out/bin/godot"
+    runHook postInstall
   '';
 
   meta = with lib; {
     homepage = "https://godotengine.org";
-    description = "Free and Open Source 2D and 3D game engine";
+    description =
+      "Free and Open Source 2D and 3D game engine" +
```

Comment:
This is overengineered. If we move the () to the description we can replace most of this with ``or ""``

Reply:
You're right, it's too much code because a just as good description format would allow for simpler code.  Simplifying.